src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -38,6 +42,11 @@
   private SSLParameters sslParameters;
   private HostnameVerifier hostnameVerifier;
 
+  private boolean isUDSConnection(String host) {
```

Comment:
I'm wondering if we might be hitting some performance issue when running this condition every time Jedis tries to establish a connection and needs to parse the supplied host. 

@allanwax @HeartSaVioR @xetorthio @sunheehnus. WDYT?